lib/media/drm_engine.js

```diff
@@ -451,26 +452,66 @@ shaka.media.DrmEngine = class {
     goog.asserts.assert(this.initialized_,
         'Must call init() before setServerCertificate');
 
-    if (this.mediaKeys_ &&
-        this.currentDrmInfo_ &&
-        this.currentDrmInfo_.serverCertificate &&
-        this.currentDrmInfo_.serverCertificate.length) {
+    if (!this.mediaKeys_ || !this.currentDrmInfo_) {
+      return;
+    }
+
+    if (this.currentDrmInfo_.serverCertificateUri &&
+        (!this.currentDrmInfo_.serverCertificate ||
+        !this.currentDrmInfo_.serverCertificate.length)) {
```

Comment:
Does the linter complain if you indent this one more space?  Normally the style is to align with the contents of the parenthesis, but our linter isn't that great about continuations.